From: Simon McVittie <smcv@collabora.com>
Date: Mon, 8 Feb 2021 10:34:50 +0000
Subject: giochannel: Don't store negative line_term_len in GIOChannel struct

Adding test coverage indicated that this was another bug in 0cc11f74.

Fixes: 0cc11f74 "giochannel: Forbid very long line terminator strings"
Bug: https://gitlab.gnome.org/GNOME/glib/-/issues/2323
Forwarded: https://gitlab.gnome.org/GNOME/glib/-/merge_requests/1931
Signed-off-by: Simon McVittie <smcv@collabora.com>
Applied-upstream: 2.66.7, commit:e069c50467712e6d607822afd6b6c15c2c343dff
---
 glib/giochannel.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/glib/giochannel.c b/glib/giochannel.c
index c3f3102..19bb06b 100644
--- a/glib/giochannel.c
+++ b/glib/giochannel.c
@@ -906,7 +906,7 @@ g_io_channel_set_line_term (GIOChannel	*channel,
 
   g_free (channel->line_term);
   channel->line_term = line_term ? g_memdup2 (line_term, length_unsigned) : NULL;
-  channel->line_term_len = length;
+  channel->line_term_len = length_unsigned;
 }
 
 /**
